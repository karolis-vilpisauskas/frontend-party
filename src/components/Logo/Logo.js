import React from "react";
import "./Logo.scss";

import black from "assets/logo-black.png";
import white from "assets/logo-white.png";

const Logo = ({ isBlack }) => {
  return (
    <img
      className={`logo ${isBlack ? "black" : "white"}`}
      src={isBlack ? black : white}
      alt="logo"
    />
  );
};

export default Logo;
