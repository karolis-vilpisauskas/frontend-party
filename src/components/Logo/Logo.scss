.logo {
  width: 80%;
  height: auto;

  &.black {
    max-width: 115px;
    max-height: 30px;
  }

  &.white {
    max-width: 246px;
    max-height: 64px;
    margin-bottom: 70px;
  }
}