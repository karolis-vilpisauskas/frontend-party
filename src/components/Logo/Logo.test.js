import '@testing-library/jest-dom/extend-expect'
import React from "react";
import { cleanup, render } from "@testing-library/react";
import Logo from "./Logo";
afterEach(cleanup);

const createTestProps = props => ({
  ...props
});

const renderTest = () => {
  const props = createTestProps();
  const { getByTestId } = render(
    <Logo {...props}>
    </Logo>
  );
  const container = getByTestId("logo");
  return {
    getByTestId,
    container
  };
};

describe("Logo", () => {
  describe("rendering", () => {
    test("it renders without crashing", () => {
      const { container } = renderTest();
      expect(container).toMatchSnapshot();
    });
  });
});
