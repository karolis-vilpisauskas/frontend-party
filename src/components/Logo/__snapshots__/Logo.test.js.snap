// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo rendering it renders without crashing 1`] = `
<img
  alt="logo"
  class="logo white"
  data-testid="logo"
  src="logo-white.png"
/>
`;
